pkg/k8s/watchers/pod.go

```diff
@@ -225,7 +225,9 @@ func (k *K8sWatcher) addK8sPodV1(pod *slim_corev1.Pod) error {
 		// called from updateK8sPodV1, the consumer will need to handle the duplicate
 		// events accordingly.
 		// GH issue #13136.
-		k.redirectPolicyManager.OnAddPod(pod)
+		if option.Config.EnableLocalRedirectPolicy {
+			k.redirectPolicyManager.OnAddPod(pod)
```

Comment:
I feel like it'd be a nice followup to get rid of all of the option checks in this file code by replacing the `k.redirectPolicyManager` with a dummy manager if the LRP is not enabled to make absolutely sure there's no paths into the LRP manager if it's disabled. But no need to address in this PR.

Reply:
Yes, I intended to add a dummy manager for unit tests too.